Packs/EclecticIQ/Integrations/EclecticIQ/EclecticIQ.py

```diff
@@ -1,794 +1,679 @@
+        return self._http_request(
+            method='POST',
+            url_suffix='/observables',
+            data=json.dumps(Body_paramas)
+        )
+
+    def get_user_granted_permissions(self) -> Any:
+        """Get user granted permissions.
+        :param: self
+        :type: str
+        :return: user granted permissions
+        :rtype: Any
+        """
+        response = self._http_request(
+            method='GET',
+            url_suffix='users/self',
+            params={}
+        )
+        data = response.get("data")
+        if data:
+            return data.get("permissions")
+        return {}
+
+    def get_platform_permissions(self) -> Any:
+        """Get platform permissions for user.
+        :param: self
+        :type: str
+        :return: permissions data
+        :rtype: Any
+        """
+        response = self._http_request(
+            method='GET',
+            url_suffix='permissions',
+            params={}
+        )
+        data = response.get("data")
+        if data:
+            return data
+        return {}
+
+
+def get_platform_permission_ids(permissions_data: Any) -> List[Any]:
+    """Get permission ids required for user to authenticate.
+    :param feeds: permissions_data
+    :type response: list
+        [{"id": 1, "name": "read history-events"},{"id": 2,"name": "read discovery-rules"}...]
+    :return: List of permission ids
+        [33, 59, 66,78]
+    :rtype: list
+    """
+    wanted_permissions = [
+        "read entitites",
+        "modify entities",
+        "read extracts",
+        "read outgoing-feeds",
+    ]
+    ids_required_for_user = []
+    for value in permissions_data:
+        if value["name"] in wanted_permissions:
+            ids_required_for_user.append(value["id"])
+
+    return ids_required_for_user
+
+
+def authenticate_user(ids_of_user: list, ids_required_for_user: list) -> Tuple[bool, List[int]]:
+    """Get user authentication and missing permission ids .
+    :param ids_of_user: permission ids user have
+    :type ids_of_user: list
+    :param ids_required_for_user: permission ids required for user to authenticate
+    :type ids_required_for_user: list
+    :return: is user authenticated , missing permissions ids
+    :rtype: boolean,list
+    """
+    user_authenticated = False
+    value = list(set(ids_required_for_user).difference(ids_of_user))
+
+    if not value:
+        user_authenticated = True
+    return user_authenticated, value
+
+
+def get_permission_name_from_id(permission_data: dict, permission_ids: list) -> Any:
+    """Get permission name from permission ids.
+    :param permission_data: permission data
+    :type permission_data: dict
+    :param permission_ids: permission id for authenticate
+    :type permission_ids: list
+    :return: permissions name
+    :rtype: Any
+    """
+    permissions_name = []
+    for data in permission_data:
+        for permission_id in permission_ids:
+            if data["id"] == permission_id:
```

Comment:
It is better to use get, in order to avoid KeyError

Reply:
Updated into get method wherever it requires.